kubectl-cloudflow/version/version.go

```diff
@@ -40,26 +36,34 @@ func FailOnProtocolVersionMismatch() {
 	if k8sErr != nil {
 		util.LogAndExit("Failed to create new kubernetes client, %s", k8sErr.Error())
 	}
-	var err error
+
 	var cm *corev1.ConfigMap
-	if cm, err = k8sClient.CoreV1().ConfigMaps(ProtocolVersionNamespace).Get(ProtocolVersionConfigMapName, metav1.GetOptions{}); err == nil {
-		operatorProtocolVersion := cm.Data[ProtocolVersionKey]
-		if operatorProtocolVersion != ProtocolVersion {
-			if version, err := strconv.Atoi(operatorProtocolVersion); err == nil {
-				if supportedVersion, err := strconv.Atoi(ProtocolVersion); err == nil {
-					if version < supportedVersion {
-						util.LogAndExit("This version of kubectl cloudflow is not compatible with the Cloudflow operator, please upgrade kubectl cloudflow")
-					}
-					if version > supportedVersion {
-						util.LogAndExit("This version of kubectl cloudflow is not compatible with the Cloudflow operator, please upgrade the Cloudflow operator")
-					}
-				}
+	if configMaps, err := k8sClient.CoreV1().ConfigMaps("").List(metav1.ListOptions{}); err == nil {
+		for _, v := range configMaps.Items {
+			if v.Name == ProtocolVersionConfigMapName {
+				cm = &v
```

Comment:
We need to change this, this actually takes the address of `v`, which will change during the iterations of the loop, so it will always point towards the last item in the slice.

We can fix this issue by instead taking the address of the item in the slice, like this:

```
	for i, v := range configMaps.Items {
			fmt.Println(v.Name)
			if v.Name == ProtocolVersionConfigMapName {
				cm = &configMaps.Items[i]
			}
		}```